### PR TITLE
[PVT-183478468] sentry

### PR DESCRIPTION
**Pull request by andyzito**
https://www.pivotaltracker.com/story/show/183478468

Port Sentry implementation from warehouse. This covers catching unhandled errors in web requests, rake tasks, and background jobs.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -39,7 +39,7 @@ gem 'memoist', require: false
 gem 'carrierwave'
 gem 'carrierwave-i18n'
 gem 'mini_magick'
-gem 'ruby-filemagic'
+# gem 'ruby-filemagic'
```
Comment (eanders) [inline Gemfile]: @andyzito any reason this is getting pulled out?

File: config/initializers/sentry.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+require_relative '../../lib/util/git'
+
+# ENV['SENTRY_DSN'] is reserved by Sentry and its use seems to prevent this initializer from being recognized.
+# Hence, we use CAS_SENTRY_DSN. Any other alternate key should also be fine.
+if ENV['CAS_SENTRY_DSN'].present?
+  Sentry.init do |config|
+    config.dsn = ENV['CAS_SENTRY_DSN']
+    config.breadcrumbs_logger = [:active_support_logger, :http_logger]
+
+    config.enabled_environments = ['production', 'staging', 'development'] # Remove development when dev is done.
```
Comment (eanders) [inline config/initializers/sentry.rb]: Do we want to remove dev now?

**Review by eanders — APPROVED** [review]
Couple of minor questions.  I'll see if I can get this up to staging sites in a bit.

File: config/initializers/sentry.rb (hunk 2 above)
Comment (andyzito) [reply config/initializers/sentry.rb]: Not if we want to capture errors from local dev environments. I put that comment in before I realized that was a use case. I'll remove it (the comment not `development`)

File: Gemfile (hunk 1 above)
Comment (andyzito) [reply Gemfile]: Cause it was preventing `bundle install` in my bare metal env and I accidentally committed it. Fixed.

**Pull request merged**